Week 04/Lecture 08/Assignment 3/crud_employee/src/main/java/com/example/crud_employee/dao/EmployeeDaoImpl.java

```diff
@@ -0,0 +1,81 @@
+package com.example.crud_employee.dao;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.example.crud_employee.model.Employee;
+
+@Repository
+public class EmployeeDaoImpl {
+
+    @Autowired
+    @Qualifier("jdbcTemplate1")
+    private JdbcTemplate jdbcTemplate1;
+
+    @Autowired
+    @Qualifier("jdbcTemplate2")
+    private JdbcTemplate jdbcTemplate2;
+
+    // Methods for datasource 1
+    public List<Employee> listEmployee1() {
+        String sql = "SELECT * FROM employee";
+        return jdbcTemplate1.query(sql, new EmployeeMapper());
+    }
+
+    public Employee getEmployee1(String id) {
+        String sql = "SELECT * FROM employee WHERE id = ?";
+        return jdbcTemplate1.queryForObject(sql, new EmployeeMapper(), id);
+    }
+
+    @Transactional(transactionManager = "dataSource1TransactionManager")
+    public void addEmployee1(Employee employee) {
+        String sql = "INSERT INTO employee (id, name, date_of_birth, address, department) VALUES (?,?,?,?,?)";
+        jdbcTemplate1.update(sql, employee.getId(), employee.getName(), employee.getDate_of_birth(), employee.getAddress(), employee.getDepartment());
+    }
+
+    @Transactional(transactionManager = "dataSource1TransactionManager")
```

Comment:
It's good to @Transaction. 1 more request - can you research what is the case @Transactional does not work.

Reply:
I have updated my lecture 08 assignment 3 for `@Transactional` does not work case.